src/rules/no-store-async.ts

```diff
@@ -0,0 +1,44 @@
+import { createRule } from "../utils"
+import type * as ESTree from "estree"
+
+export default createRule("no-store-async", {
+  meta: {
+    docs: {
+      description:
+        "disallow using async/await inside svelte stores because it causes issues with the auto-unsubscribing features",
+      category: "Possible Errors",
+      recommended: true,
+      default: "error",
+    },
+    schema: [],
+    messages: {
+      unexpected: "Do not pass async functions to svelte stores.",
+    },
+    type: "problem",
+  },
+  create(context) {
+    return {
+      CallExpression(node: ESTree.CallExpression) {
+        if (node.callee.type !== "Identifier") return
+        const { name } = node.callee
+        if (name !== "writable" && name !== "readable" && name !== "derived")
```

Comment:
Could you change it to use eslint-utils? By using eslint-utils, we can also check for cases like:

```js
import * as stores from "svelte/store"
stores.writable()
```

https://eslint-utils.mysticatea.dev/api/scope-utils.html#referencetracker-class

I think the implementation of require-stores-init is helpful.
https://github.com/ota-meshi/eslint-plugin-svelte/blob/3362ad4053c554ad54f7c1e42193282f7d2fdaac/src/rules/require-stores-init.ts#L21-L35

Reply:
updated!

https://github.com/ota-meshi/eslint-plugin-svelte/pull/225/commits/37d1de783bd7a6897a5f1b7377897576129ad485

I moved `extractStoreReferences` to utils folder but I'm not sure if this is your preferred way.